File: hw/palomo/u3/hw24CompositePattern/Composite/src/main/java/ec/edu/espe/composite/model/Employee.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.composite.model;

/**
 *
 * @author Cristian Palomo, Error 404, @ESPE
 */
public abstract class Employee {
    
    protected final String name;
    
    protected Employee(String name){
        this.name = name;
    }
    
    protected abstract String getRole();
    
    public void stateName(){
        System.out.println(name + "-" + getRole());
    }
    
    public String getName(){
        return name;
    }
}

File: hw/palomo/u3/hw24CompositePattern/Composite.ts/src/model/Employee.ts
export abstract class Employee {
    protected name: string;

    protected constructor(name: string) {
        this.name = name;
    }

    protected abstract getRole(): string;

    stateName(): void {
        console.log(`${this.name} - ${this.getRole()}`);
    }

    getName(): string {
        return this.name;
    }
}